lectures/04_lecture/data_pipelines.ipynb

```diff
@@ -0,0 +1,1576 @@
+{
```

Comment:
Issues:
- a lot of unused imports -> remove them
- imports in 3 cells at the beginning - merge into one
- there are a few typos -> check the texts
- use semicolons when plotting 
- stick to one style of comments - capital letters at the beginning vs. lowercase
- minimize warnings